airflow/providers/microsoft/azure/secrets/key_vault.py

```diff
@@ -127,7 +134,13 @@ def client(self) -> SecretClient:
         if all([self.tenant_id, self.client_id, self.client_secret]):
             credential = ClientSecretCredential(self.tenant_id, self.client_id, self.client_secret)
         else:
-            credential = DefaultAzureCredential()
+            if all([self.managed_identity_client_id, self.workload_identity_tenant_id]):
+                credential = DefaultAzureCredential(
```

Comment:
Shall we cover these changes under test?

Reply:
Yes, will add some tests tomorrow. Thanks for reminding me :)

Reply:
```suggestion
         if self.managed_identity_client_id and self.workload_identity_tenant_id:
                credential = DefaultAzureCredential(
```
Since it's just two values we need to check.

Reply:
@Adaverse just added the unit tests. thanks!

@utkarsharma2 makes sense. just updated it. thanks!